lib/rubocop/cop/bundler/gem_filename.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Bundler
+      # This cop verifies that a project contains Gemfile or gems.rb file and correct
+      # associated lock file based on the configuration.
+      #
+      # @example RequiresGemfile: true (default)
+      #   # bad
+      #   Project contains gems.rb and gems.locked files
+      #
+      #   # bad
+      #   Project contains Gemfile and gems.locked file
+      #
+      #   # good
+      #   Project contains Gemfile and Gemfile.lock
+      #
+      # @example RequiresGemfile: false
+      #   # bad
+      #   Project contains Gemfile and Gemfile.lock files
+      #
+      #   # bad
+      #   Project contains gems.rb and Gemfile.lock file
+      #
+      #   # good
+      #   Project contains gems.rb and gems.locked files
+      class GemFilename < Base
+        include RangeHelp
+
+        MSG_GEMFILE_REQUIRED = 'gems.rb file was found but Gemfile is required.'
```

Comment:
Please add backticks around the names of the files, so they are properly colored in the terminal.